### PR TITLE
[proof] add comments in verify_sparse_merkle_element() for readability

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation
To improve readability.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?
yes

## Test Plan

existing coverage

### CODE REVIEW

File: types/src/proof/mod.rs [hunk 1]
```diff
@@ -453,10 +457,15 @@ pub fn verify_sparse_merkle_element(
...
+        (None, None) => {
+            // This is a non-inclusion proof.
+            // The proof intends to show that if a leaf node representing `element_key` is inserted,
+            // it will show up at a currently empty position.
+            // `sibling` should prove the route from this empty position to the root.
+        }
     }
 
-    let leaf_hash = match sparse_merkle_proof.leaf() {
+    let current_hash = match sparse_merkle_proof.leaf() {
```
Comment (wqfish) [inline types/src/proof/mod.rs]: @msmouse why change this one? It is the leaf hash.

File: types/src/proof/mod.rs [hunk 2]
```diff
@@ -469,7 +478,7 @@ pub fn verify_sparse_merkle_element(
                 .rev()
                 .skip(HashValue::LENGTH_IN_BITS - siblings.len()),
         )
-        .fold(leaf_hash, |hash, (sibling_hash, bit)| {
+        .fold(current_hash, |hash, (sibling_hash, bit)| {
```
Comment (wqfish) [inline types/src/proof/mod.rs]: I think

```
.fold(leaf_hash, |current_hash, (sibling_hash, bit)| {
```

would be good.

Comment (zamsden) [reply types/src/proof/mod.rs]: That does seem slightly better.

File: types/src/proof/mod.rs (hunk 1 above)
Comment (lightmark) [reply types/src/proof/mod.rs]: I think @msmouse would argue some "leaf" is not real leaf but just placeholder hash representing an empty subtree.

**Pull request merged**